### DISCUSSION TITLE
Once the bot is running, is the bot supposed to display messages sent/received while bot was offline ?

### DISCUSSION THREAD

**Discussion by Cm79** [FKLC/WhatsAppToDiscord#177] [Q&A]
Hello, 

Currently the wa2dc bot only displays messages it received/sent while it was running. 
I was wondering why it has this behavior, maybe I installed incorrectly, or it is a feature yet to be developed,
or it is a feature too complicated/impossible to develop ?

I see some people are hosting the bot,, I don't really need the bot to run 24/7, that's why I ask.

Thanks

**Comment by FKLC** [comment]
Any service handing stuff all the time, has a server 7/24 running. Even the so called "serverless" services use servers. I know where you come from though, generally you don't run bots, but just invite them, but in the background there's someone hosting the bot in a server. So, it is Impossible to handle messages/logic/commands without the bot running on a some sort of computer 7/24 if you want it to run 7/24.

**Reply by Cm79** [reply]
Thanks for the reply, I understand that it's impossible for the bot to distribute messages in real time if the bot is not running somewhere.

I was thinking of a function that would run when the bot launches, the function would check for each conversation if the last message on Discord corresponds to the last message on Whatsapp, and if not it would send the missing messages.

I don't know if it's technically possible with Baileys and discord.js though.
Also, I sometimes see this line in my logs when I launch the wa2dc application: `"version":"v0.10.24","msg":"handled 1 offline messages/notifications"`, does it correspond to anything ? I feel like it's not generated directly by the wa2dc code but by a lib.
